daemon/src/collab_settlement/maker.rs

```diff
@@ -0,0 +1,264 @@
+                let mut framed =
+                    Framed::new(stream, JsonCodec::<ListenerMessage, DialerMessage>::new());
+
+                let propose = framed
+                    .next()
+                    .await
+                    .context("End of stream while receiving Propose")?
+                    .context("Failed to decode Propose")?
+                    .into_propose()?;
+
+                address
+                    .send(ProposeReceived {
+                        propose,
+                        framed,
+                        peer,
+                    })
+                    .await?;
+
+                anyhow::Ok(())
+            },
+            |e| async move { tracing::warn!("Failed to handle incoming collab settlement: {e:#}") },
+        );
+    }
+}
+
+#[xtra_productivity]
+impl Actor {
+    async fn handle(&mut self, msg: ProposeReceived) {
+        let ProposeReceived {
+            propose,
+            framed,
+            peer,
+        } = msg;
+        let order_id = propose.id;
+
+        let result = self
+            .executor
+            .execute(order_id, |cfd| {
+                cfd.verify_counterparty_peer_id(&peer.into())?;
+                cfd.start_collab_settlement_maker(
+                    propose.price,
+                    self.n_payouts,
+                    &propose.unsigned_tx,
+                )
+            })
+            .await
+            .context("Failed to start collab settlement protocol");
+
+        let (transaction, proposal) = match result {
+            Ok((transaction, proposal)) => (transaction, proposal),
+            Err(e) => {
+                tracing::debug!(%order_id, %peer, "Failed to start collab settlement protocol: {e:#}");
+                emit_failed(order_id, e, &self.executor).await;
+                return;
+            }
+        };
+
+        self.pending_protocols
+            .insert(order_id, (framed, transaction, proposal, peer));
+    }
+
+    async fn handle(&mut self, msg: Accept) -> Result<()> {
+        let Accept { order_id } = msg;
+
+        let (mut framed, transaction, proposal, _peer) =
+            self.pending_protocols
+                .remove(&order_id)
+                .with_context(|| format!("No active protocol for order {order_id}"))?;
+
+        let mut tasks = Tasks::default();
+        tasks.add_fallible(
+            {
+                let executor = self.executor.clone();
+                async move {
+                    executor
+                        .execute(order_id, |cfd| {
+                            cfd.accept_collaborative_settlement_proposal(&proposal)
+                        })
+                        .await?;
+
+                    framed
+                        .send(ListenerMessage::Msg1(Decision::Accept))
+                        .await
+                        .context("Failed to send Msg1::Accept")?;
+
+                    // TODO: We will need to apply a timeout to these. Perhaps we can put a timeout
```

Comment:
I'd just add the timeout and remove the Todo.

Reply:
I'd prefer to merge this body of work and then deal with it in #2156 due to potential of splitting remaining work.